src/rgw/rgw_admin.cc

```diff
@@ -1402,6 +1411,42 @@ int set_bucket_quota(rgw::sal::Driver* driver, OPT opt_cmd,
   return 0;
 }
 
+int is_versioned_instance_listable(RGWRados* store,
+                                   RGWRados::BucketShard& bs,
+                                   const cls_rgw_obj_key& key,
+                                   bool& listable) {
+  string marker = "";
+  list<rgw_cls_bi_entry> entries;
+  bool is_truncated;
+  listable = false;
+  int ret;
+  do {
+    ret = store->bi_list(bs, key.name, marker, 1000, &entries, &is_truncated);
```

Comment:
consider using `cls_rgw_bucket_list_op()` instead, which should only be returning the plain entries. `bi_list()` would search each of the special namespaces too

Reply:
I'm breaking out of the loop when it encounters a non-plain entry, so it won't actually iterate over the special namespaces. The issue with using `cls_rgw_bucket_list_op` is that it's susceptible to high latencies and errors when a bunch of these bogus entries are in the index (the ones that we're trying to clean up). That said, I acknowledge that I do need to do something different here because breaking out of the loop when I hit the first non-plain entry means that I'm not considering any plain entries that come after the special namespaces. I'll review this.